ros_ws/src/autonomous/evolutionary/scripts/train.py

```diff
@@ -18,24 +22,34 @@ def __init__(self):
         rospy.Subscriber(TOPIC_CRASH, Empty, self.on_crash)
 
         self.population = []
-        self.untested_population = [NeuralCarDriver()
-                                    for _ in range(POPULATION_SIZE)]
-        self.current_driver = self.untested_population[0]
 
         self.is_terminal_step = False
         self.episode_length = 0
         self.generation = 0
         self.test = 0
 
+        self.untested_population = []
+        self.spawn_forward = True
+        if CONTINUE_TRAINING:
+            for i in range(POPULATION_SIZE):
+                individuum = NeuralCarDriver()
+                individuum.load(i)
+                self.untested_population.append(individuum)
+        else:
+            self.untested_population = [NeuralCarDriver()
+                                        for _ in range(POPULATION_SIZE)]
```

Comment:
Here, you can replace `POPULATION_SIZE` with `INITIAL_RANDOM_GENERATION * POPULATION_SIZE`.
That way, you no longer need all the checks if we are in the initial random generation. You can rename it to `INITIAL_POPULATION_SIZE` to simplify it even more.